src/brevitas/graph/equalize.py

```diff
@@ -1326,54 +1350,110 @@ def _apply_rotate(model: nn.Module, regions: List[Region], full_rotation_method=
                 print("Skipping layers")
                 continue
 
+        # If the rotation is not fused, redefine as a Parameter, to enable its optimization
+        if not insert_rotation_module and not fuse_rotations:
+            rot_mat = torch.nn.Parameter(rot_mat)
+
         for name, indexes in region.srcs.items():
             module = region.get_module_from_name(name)
-            if hasattr(module, 'allocate_params'):
-                module.allocate_params(module)
             axis = _get_output_axis(module)
-            weight = module.weight.data
 
-            if axis == 0:
-                weight = rot_func(weight.t(), rot_mat, K).t()
-            elif axis == 1:
-                weight = rot_func(weight, rot_mat, K)
-            else:
-                raise RuntimeError("Not supported yet")
-            module.weight.data = weight
+            if fuse_rotations:
+                rewriter = ModuleInstanceFuseRotationWeights(
```

Comment:
If this is equivalent to the old behavior, it seems a bit over-complicated. Do we need it?